File: sources/paopao/Kart/Physics/Suspension.cs
using System;
using UnityEngine;

public struct Suspension
{
	public void InitVector2(ref Vector2 v, float x, float y)
	{
		v.x = x;
		v.y = y;
	}

	public void Initialize()
	{
		this.wheelOff = new Vector2[4];
		for (int i = 0; i < 4; i++)
		{
			this.InitVector2(ref this.wheelOff[i], (i % 2 != 0) ? 1f : (-1f), (i / 2 != 0) ? (-1f) : 1f);
		}
		this.wheelContact = new bool[4];
		this.wheelContactN = new Vector3[4];
		this.maxTravel = 0.3f;
		this.travel = new float[] { 0.3f, 0.3f, 0.3f, 0.3f };
		this.deltaTravel = new float[4];
	}

	public Vector2[] wheelOff;

	public Vector3 contactN;

	public bool[] wheelContact;

	public Vector3[] wheelContactN;

	public float maxTravel;

	public float[] travel;

	public float[] deltaTravel;
}

File: sources/roblox-lua-items-network/shared/kartEngine/KartMove/Suspension.lua
-- Suspension struct的Lua等效实现
local UnityMath = require(script.Parent.Parent.KartShared.UnityMath)
local Vector2 = UnityMath.Vector2
local Vector3 = UnityMath.Vector3

local Suspension = {}
Suspension.__index = Suspension

function Suspension.new()
    local self = setmetatable({}, Suspension)
    self:Initialize()
    return self
end

function Suspension:InitVector2(v, x, y)
    v.X = x
    v.Y = y
end

function Suspension:Initialize()
    self.wheelOff = {}
    for i = 1, 4 do
        self.wheelOff[i] = Vector2.new()
        local xVal = ((i - 1) % 2 ~= 0) and 1.0 or -1.0
        local yVal = (math.floor((i - 1) / 2) ~= 0) and -1.0 or 1.0
        self:InitVector2(self.wheelOff[i], xVal, yVal)
    end
    
    self.wheelContact = {}
    for i = 1, 4 do
        self.wheelContact[i] = false
    end
    
    self.wheelContactN = {}
    for i = 1, 4 do
        self.wheelContactN[i] = Vector3.new()
    end
    
    self.maxTravel = 1.5  -- 大幅增加悬挂行程，让车体离地更高
    self.travel = {1.5, 1.5, 1.5, 1.5}
    self.deltaTravel = {0.0, 0.0, 0.0, 0.0}
    
    -- 添加遗漏的contactN字段
    self.contactN = Vector3.new()
end

return Suspension